### DISCUSSION TITLE
What is Hubble?

### DISCUSSION THREAD

**Discussion by roopeshsn** [microsoft/retina#546] [Q&A] [answered]
I have noticed some issues on Hubble. Also, I am not able to find relevant documentation for it. What is Hubble?

**Comment by rbtr** [comment]
https://github.com/cilium/hubble?tab=readme-ov-file#what-is-hubble

**Reply by roopeshsn** [reply]
So depending on the choice the metrics can be exported to Hubble instead of Grafana right?

**Comment by SRodi** [comment]
@kamilprz

**Comment by kamilprz** [comment, marked as answer]
Hubble is a networking and security **observability platform** designed for cloud-native workloads. It’s built on top of Cilium and eBPF, which allows it to provide deep visibility into the communication and behavior of services and the networking infrastructure.

You can read more here: https://docs.cilium.io/en/stable/overview/intro/#what-is-hubble

Retina supports using Hubble as the control plane without the need for Cilium CNI. There is also a non-Hubble control plane (legacy).

Regardless of the control plane, metrics are exposed to Prometheus endpoints (and thus can be visualised in Grafana). If the Hubble control plane is used, metrics are also sent to Hubble.

**Reply by kamilprz** [reply]
@roopeshsn let me know if this helps :)

**Reply by kamilprz** [reply]
p.s. we have also opened an issue to add more info about Hubble to our docs https://github.com/microsoft/retina/issues/1055

**Reply by SRodi** [reply]
We are in the process of updating the docs to better describe Hubble control plane for Retina. Some initial work was published today to provide a list of available metrics for Hubble control plane https://retina.sh/docs/Metrics/metrics-intro

**Reply by gespi1** [reply]
Also the install details for the retina-hubble chart if there even is one? Looks like the installation instructions only mention legacy. If not, is a chart incoming?

**Reply by SRodi** [reply]
Hi @gespi1, I am currently working on a PR to publish Hubble control plane chart and update the installation documents.  https://github.com/microsoft/retina/pull/1223

**Reply by kamilprz** [reply]
We also have a PR open to add some documentation to `retina.sh` about our Data Plane / Control Planes, including Hubble :) https://github.com/microsoft/retina/pull/1229
